src/examples/aes/common/aescircuit.cpp

```diff
@@ -45,16 +45,22 @@ int32_t test_aes_circuit(e_role role, const std::string& address, uint16_t port,
 	verify.Create(AES_BITS * nvals);
 	key.CreateBytes(AES_EXP_KEY_BYTES);
 
-
-
-	//TODO create random key and perform key schedule, right now a static (expanded) key is used
-	key.Copy((uint8_t*) AES_TEST_EXPANDED_KEY, 0, AES_EXP_KEY_BYTES);
+	uint8_t aes_test_key[AES_KEY_BYTES];
+	srand(7438);
+	for(uint32_t i = 0; i < AES_KEY_BYTES; i++) {
+		aes_test_key[i] = (uint8_t) (rand() % 256);
+	}
```

Comment:
Before it was a static (expanded) key and now the key is generated using a static seed. So it is not *more* random than before.

Reply:
That is true, but the test can't be "real" random since the input depends on both parties and the parties must agree to some common input before the SFE, otherwise the verification process would never work.

My only suggestions would be to 
(1) Make the seed an input of the function and let the user provide a seed rather than just use a hardcoded seed.
(2) Create a process which makes it possible to share common information before starting creating the funcion. That would be a feature I would like to be working on soon anyways.

Reply:
I agree, for testing you have to verify the correctness of the computation somehow ...

The solution with the C random number generator (and static seed) is not better than just putting a static AES key in the source (now in non-expanded form).